CONTRIBUTING.md

```diff
@@ -57,13 +56,62 @@ If you want to amend an existing changelog entry part of a PR already merged on
 
       CHANGELOG_END
 
+If the PR contains no _user-facing_ change, the section **MUST** be there but can be left empty, as in the following example:
+
+      Fixes #6666
+
+      Improve contribution guidelines
+
+      CHANGELOG_BEGIN
+      CHANGELOG_END
+
 If you want to verify the changelog entries as described by a range of Git revisions, you can use the `unreleased.sh` script. In most cases, to see the entries added as part of commits added since branching off of `master`, you can run:
 
     ./unreleased.sh master..
 
+## Writing a good changelog entry
+
+Writing good changelog entries is **important**: as a developer, it gives visibility on your contribution; as a user, it makes clear what is new, what's changed, and how to deal with them, making the product more accessible and your work more meaningful.
+
+The raw changelog is used to compile a meaningful summary of changes across releases. This happens some time after the PR has been merged and the person taking the responsibility of summarizing new user-facing features must be in the position to easily understand the nature of the change and report it. The ideal changelog entry can be more or less incorporated verbatim in the release notes.
+
+Here are a few practical tips:
+
+* if there are no user-facing changes, keep the changelog entry list empty
+* the first term to appear should be the affected component -- [here's a list](#list-of-components-for-changelog-entries)
+* write as many changelog entries as necessary
+* don't be _too_ succinct: a single entry does **not have to** fit on a single line
+* on the other end, if the size grows beyond 5-6 lines, rather add a link to a relevant documentation or issue with more details
+* the ultimate target are end users: focus on the impact on them, tell them what's new or how to deal with a change
+
+### List of components for changelog entries
+
+  * Compiler
+  * DAML Maven Plugin
+  * DAML on SQL
+  * DAML Studio
+  * Examples & Docs
+  * Extractor
+  * Java Bindings
+  * Java Codegen
+  * JavaScript Client Libraries
+  * JSON API
+  * Ledger API Specification
+  * Ledger API Test Tool
+  * Navigator
+  * REPL
+  * Sandbox
+  * Scala Bindings
+  * Scala Codegen
+  * Script
+  * SDK Assistant
+  * Standard Library
+  * Test Library
+  * Triggers
```

Comment:
```suggestion
  * DAML Compiler
  * DAML on SQL
  * DAML Studio
  * Extractor
  * Java Bindings
  * Java Codegen
  * JavaScript Client Libraries
  * JavaScript Codegen
  * JSON API
  * Ledger API Specification
  * Integration Kit (Ledger API Test Tool and changes only relevant for ledger integrations, e.g, kvutils changes, metrics, …)
  * Navigator
  * DAML REPL
  * Sandbox
  * Scala Bindings
  * Scala Codegen
  * DAML Script
  * DAML Assistant
  * DAML Standard Library
  * DAML Triggers
```

Reply:
Looks pretty comprehensive. I wouldn't treat this as an exhaustive list but rather as guidance of the kind of thing that goes in there. This should cover 95%+ of all changelog entries. One suggestion:

Maybe "Distribution/Releases" would be something to include for changes like releasing the JSON API to GH Releases.